### PR TITLE
feat: add workflow dispatch and better logging level

### PR DESCRIPTION
**Pull request by himanshu-dixit**
> [!IMPORTANT]
> Adds workflow dispatch for manual releases and improves logging levels across the SDK.
> 
>   - **Workflow Dispatch**:
>     - Adds `workflow_dispatch` event to `release_js_sdk.yml` for manual triggering with inputs `version` and `isRCRelease`.
>     - Updates version using `npm version` and conditionally tags as `rc` during `pnpm publish` if `isRCRelease` is true.
>   - **Logging**:
>     - Changes logging level from `info` to `debug` in `workspace.ts`, `Entity.ts`, and `triggers.ts`.
>     - Updates `logger.ts` to define log levels and colors, and configures log format.
>     - Enhances logging in `config.ts` to include request and response details with timing.
>   - **Misc**:
>     - Adds `getLogLevel()` export in `logger.ts` for retrieving log level from environment variables.
>     - Refactors `getAPISDK()` to `getOpenAPIClient()` in multiple files for consistency.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=ComposioHQ%2Fcomposio&utm_source=github&utm_medium=referral)<sup> for 51770d76f7f7731afe8a46f4735f4060a62badea. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

File: .github/workflows/release_js_sdk.yml [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
-    # For example, 'js-v1.0.0' or 'js-release-2.3.1'
-    tags:        
-      - 'js-*'  
-    
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to publish (e.g., 1.0.0)'
+        required: true
+        type: string
```
Comment (shreysingla11) [inline .github/workflows/release_js_sdk.yml:9]: Consider adding pattern validation for the version input to ensure it follows semver format:
```yaml
version:
  description: 'Version to publish (e.g., 1.0.0)'
  required: true
  type: string
  pattern: '^[0-9]+\.[0-9]+\.[0-9]+(-rc\.[0-9]+)?$'
```

File: .github/workflows/release_js_sdk.yml [hunk 2]
```diff
@@ -35,9 +38,14 @@ jobs:
       env:
         NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
 
+    - name: Update version
+      run: npm version ${{ github.event.inputs.version }} --no-git-tag-version
```
Comment (shreysingla11) [inline .github/workflows/release_js_sdk.yml:42]: The npm version command should include git configuration to ensure proper commit author:
```yaml
- name: Update version
  run: |
    git config --global user.email "github-actions@github.com"
    git config --global user.name "GitHub Actions"
    npm version ${{ github.event.inputs.version }} --no-git-tag-version
```

File: .github/workflows/release_js_sdk.yml [hunk 3]
```diff
@@ -35,9 +38,14 @@ jobs:
...
+
+    - name: Commit and push
+      run: git commit -am "Release ${{ github.event.inputs.version }}" && git push
```
Comment (shreysingla11) [inline .github/workflows/release_js_sdk.yml:45]: The git commit command should handle cases where there are no changes to commit:
```yaml
- name: Commit and push
  run: |
    if git diff --quiet; then
      echo "No changes to commit"
    else
      git commit -am "Release ${{ github.event.inputs.version }}" && git push
    fi
```

File: .github/workflows/release_js_sdk.yml [hunk 4]
```diff
@@ -35,9 +38,14 @@ jobs:
...
+
     - name: pnpm build
       run: pnpm build
       
     - name: pnpm publish
       run: pnpm publish --no-git-checks
```
Comment (shreysingla11) [inline .github/workflows/release_js_sdk.yml:51]: Consider using the isRCRelease input to modify the npm publish command for RC releases:
```yaml
- name: pnpm publish
  run: pnpm publish --no-git-checks ${{ github.event.inputs.isRCRelease && '--tag rc' || '' }}
```

**Comment by shreysingla11** [comment]
### Code Review Summary

The changes to add workflow dispatch for JS SDK releases are generally good, but there are several areas that could be improved for better reliability and maintainability:

**Strengths:**
✅ Good transition from tag-based to manual workflow dispatch
✅ Addition of RC release support
✅ Proper use of secrets and environment variables

**Suggestions for Improvement:**
1. Add version input validation using regex pattern
2. Include proper git configuration for version updates
3. Add error handling for git commit step
4. Implement RC tag handling in publish step
5. Consider adding a step to create git tags for releases

**Overall Assessment:**
The changes are moving in the right direction but need some refinements for robustness. The core functionality is solid, but adding the suggested improvements would make the workflow more reliable and maintainable.

Rating: 7/10 - Good foundation but needs additional error handling and validation.

File: .github/workflows/release_js_sdk.yml (hunk 1 above)
Comment (plxity) [reply .github/workflows/release_js_sdk.yml:9]: @himanshu-dixit I believe, it's good to add a pattern check. what do you think?

**Review by plxity — APPROVED** [review]

File: .github/workflows/release_js_sdk.yml (hunk 1 above)
Comment (himanshu-dixit) [reply .github/workflows/release_js_sdk.yml:9]: AI hallucination, pattern is not supported
Screenshot 2024-11-29 at 1 18 43 PM

Comment (plxity) [reply .github/workflows/release_js_sdk.yml:9]: lol my bad. LGTM

**Pull request merged**